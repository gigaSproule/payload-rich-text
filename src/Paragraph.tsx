import type { TextData } from "./Text";
import { Text } from "./Text";
import type { LinkData } from "./Link";
import { Link } from "./Link";
import type { Options } from "./types";
import type { InlineBlockData } from "./InlineBlock";
import { InlineBlock } from "./InlineBlock";
import type { CSSProperties } from "react";
import { LineBreak } from "./LineBreak";
import type { LineBreakData } from "./LineBreak";

export interface ParagraphData {
  type: "paragraph";
  children: (TextData | LinkData | LineBreakData | InlineBlockData)[];
  version: number;
  indent: number;
  format: "left" | "start" | "center" | "right" | "end" | "justify" | "";
  direction: "ltr" | "rtl" | null;
  // TODO: These match what is passed into Text children, but InlineBlock doesn't have a format nor style property
  textFormat: number;
  textStyle: string;
}

export type Props = { data: ParagraphData; options?: Options };

const getStyle = (data: ParagraphData): CSSProperties | undefined => ({
  textAlign: data.format !== "" ? data.format : undefined,
  direction: data.direction === "rtl" ? data.direction : undefined,
});

export const Paragraph = ({ data, options }: Props) => {
  const style: CSSProperties | undefined = getStyle(data);
  const children = data.children.map((paragraphChild, index) => {
    if (paragraphChild.type === "text") {
      return (
        <Text
          key={paragraphChild.text}
          data={paragraphChild}
          options={options}
          style={style}
        />
      );
    } else if (paragraphChild.type === "link") {
      return (
        <Link
          key={paragraphChild.id}
          data={paragraphChild}
          options={options}
          style={style}
        />
      );
    } else if (paragraphChild.type === "linebreak") {
      return <LineBreak key={index} data={paragraphChild} options={options} />;
    } else {
      return (
        <InlineBlock
          key={index}
          data={paragraphChild}
          options={options}
          style={style}
        />
      );
    }
  });
  return options?.paragraph ? (
    options.paragraph(data, children)
  ) : (
    <p>{children}</p>
  );
};
