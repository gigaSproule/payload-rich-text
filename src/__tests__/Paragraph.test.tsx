import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../Paragraph";
import { Paragraph } from "../Paragraph";
import { createTextData } from "./helpers/createTextData";
import { createLinkData } from "./helpers/createLinkData";
import { createInlineBlockData } from "./helpers/createInlineBlockData";
import { createParagraphData } from "./helpers/createParagraphData";

describe("Paragraph", () => {
  it("should not render anything if no children are provided", () => {
    const paragraphData: Props["data"] = createParagraphData({
      children: [],
    });
    const { container } = render(<Paragraph data={paragraphData} />);
    expect(container).toMatchSnapshot();
  });

  it(`should render default text if no options provided`, () => {
    const paragraphData: Props["data"] = createParagraphData();
    const { container } = render(<Paragraph data={paragraphData} />);
    expect(container).toMatchSnapshot();
  });

  (
    [
      "left",
      "start",
      "center",
      "right",
      "end",
      "justify",
      "",
    ] as Props["data"]["format"][]
  ).forEach((format) =>
    (["ltr", "rtl", null] as Props["data"]["direction"][]).forEach(
      (direction) =>
        it(`should render default text with correct styling if no options provided and format is ${format} and direction is ${direction}`, () => {
          const paragraphData: Props["data"] = createParagraphData({
            children: [createTextData()],
            direction: direction,
            format: format,
          });
          const { container } = render(<Paragraph data={paragraphData} />);
          expect(container).toMatchSnapshot();
        }),
    ),
  );

  (
    [
      "left",
      "start",
      "center",
      "right",
      "end",
      "justify",
      "",
    ] as Props["data"]["format"][]
  ).forEach((format) =>
    (["ltr", "rtl", null] as Props["data"]["direction"][]).forEach(
      (direction) =>
        it(`should render default link with correct styling if no options provided and format is ${format} and direction is ${direction}`, () => {
          const paragraphData: Props["data"] = createParagraphData({
            children: [createLinkData()],
            direction: direction,
            format: format,
          });
          const { container } = render(<Paragraph data={paragraphData} />);
          expect(container).toMatchSnapshot();
        }),
    ),
  );

  (
    [
      "left",
      "start",
      "center",
      "right",
      "end",
      "justify",
      "",
    ] as Props["data"]["format"][]
  ).forEach((format) =>
    (["ltr", "rtl", null] as Props["data"]["direction"][]).forEach(
      (direction) =>
        it(`should render default inline block with correct styling if no options provided and format is ${format} and direction is ${direction}`, () => {
          const paragraphData: Props["data"] = createParagraphData({
            children: [createInlineBlockData()],
            direction: direction,
            format: format,
          });
          const { container } = render(<Paragraph data={paragraphData} />);
          expect(container).toMatchSnapshot();
        }),
    ),
  );

  (
    [
      "left",
      "start",
      "center",
      "right",
      "end",
      "justify",
      "",
    ] as Props["data"]["format"][]
  ).forEach((format) =>
    (["ltr", "rtl", null] as Props["data"]["direction"][]).forEach(
      (direction) =>
        it(`should render custom render with correct styling if options provided with paragraph and format is ${format} direction is ${direction}`, () => {
          const paragraphData: Props["data"] = createParagraphData({
            direction: direction,
            format: format,
          });
          const { container } = render(
            <Paragraph
              data={paragraphData}
              options={{
                paragraph: (paragraph, children) => {
                  return (
                    <div>
                      <span>
                        Custom Paragraph - {JSON.stringify(paragraph)}
                      </span>
                      {children}
                    </div>
                  );
                },
              }}
            />,
          );
          expect(container).toMatchSnapshot();
        }),
    ),
  );
});
