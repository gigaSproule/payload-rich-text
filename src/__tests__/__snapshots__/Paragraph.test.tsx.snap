// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should not render anything if no children are provided 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is  direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span>
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is  direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span>
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is  direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is center direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"center","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: center;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is center direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"center","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: center;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is center direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"center","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: center; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is end direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"end","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: end;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is end direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"end","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: end;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is end direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"end","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: end; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is justify direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"justify","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: justify;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is justify direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"justify","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: justify;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is justify direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"justify","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: justify; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is left direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"left","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: left;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is left direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"left","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: left;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is left direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"left","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: left; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is right direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"right","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: right;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is right direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"right","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: right;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is right direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"right","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: right; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is start direction is ltr 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"ltr","format":"start","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: start;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is start direction is null 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":null,"format":"start","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: start;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render custom render with correct styling if options provided with paragraph and format is start direction is rtl 1`] = `
<div>
  <div>
    <span>
      Custom Paragraph - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"direction":"rtl","format":"start","indent":0,"type":"paragraph","version":1,"textFormat":0,"textStyle":""}
    </span>
    <span
      style="text-align: start; direction: rtl;"
    >
      heading text
    </span>
  </div>
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is  and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is  and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is  and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is center and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is center and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is center and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is end and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is end and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is end and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is justify and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is justify and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is justify and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is left and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is left and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is left and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is right and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is right and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is right and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is start and direction is ltr 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is start and direction is null 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default inline block with correct styling if no options provided and format is start and direction is rtl 1`] = `
<div>
  <p />
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is  and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is  and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is  and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is center and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: center;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is center and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: center;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is center and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: center; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is end and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: end;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is end and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: end;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is end and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: end; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is justify and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: justify;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is justify and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: justify;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is justify and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: justify; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is left and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: left;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is left and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: left;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is left and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: left; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is right and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: right;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is right and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: right;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is right and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: right; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is start and direction is ltr 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: start;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is start and direction is null 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: start;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default link with correct styling if no options provided and format is start and direction is rtl 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
      style="text-align: start; direction: rtl;"
    >
      heading text
    </a>
  </p>
</div>
`;

exports[`Paragraph should render default text if no options provided 1`] = `
<div>
  <p>
    <span>
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is  and direction is ltr 1`] = `
<div>
  <p>
    <span>
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is  and direction is null 1`] = `
<div>
  <p>
    <span>
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is  and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is center and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: center;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is center and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: center;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is center and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: center; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is end and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: end;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is end and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: end;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is end and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: end; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is justify and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: justify;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is justify and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: justify;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is justify and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: justify; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is left and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: left;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is left and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: left;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is left and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: left; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is right and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: right;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is right and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: right;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is right and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: right; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is start and direction is ltr 1`] = `
<div>
  <p>
    <span
      style="text-align: start;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is start and direction is null 1`] = `
<div>
  <p>
    <span
      style="text-align: start;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render default text with correct styling if no options provided and format is start and direction is rtl 1`] = `
<div>
  <p>
    <span
      style="text-align: start; direction: rtl;"
    >
      heading text
    </span>
  </p>
</div>
`;

exports[`Paragraph should render linebreak if children has inlineblock 1`] = `
<div>
  <p>
    <div>
      <span>
        Custom InlineBlock - 
        {"type":"inlineBlock","version":1,"fields":{"blockType":"some-block-type","some":"custom","block":"fields"}}
      </span>
      {}
    </div>
  </p>
</div>
`;

exports[`Paragraph should render linebreak if children has linebreak 1`] = `
<div>
  <p>
    <br />
  </p>
</div>
`;

exports[`Paragraph should render linebreak if children has link 1`] = `
<div>
  <p>
    <a
      href="https://www.example.com"
    >
      heading text
    </a>
  </p>
</div>
`;
